tests/common.py

```diff
@@ -99,8 +99,8 @@ def run(
         del os.environ["PYTHONPATH"]
 
     results_dir = join(path, "expected-results")
-    snakefile = join(path, snakefile)
-    assert os.path.exists(snakefile)
```

Comment:
Maybe the failure is a side effect of your changes here?

Reply:
My changes there honor using the Snakefile in the working directory - without that it uses the Snakefile in the original directory (and this breaks my previously working test). Isn’t the fixed approach I did more correct?

Reply:
@johanneskoester it shouldn't be the case that a job running from a working directory in /tmp is using a snakefile that is in the original folder. Or further, it shouldn't be the case that changing the path from the original working directory to the staged temp should break a test (as it did this one). From the test for google life sciences point of view, it breaks because we have to honor the path of the Snakefile (relative to the working directory) to build the .tar.gz package. So if we build with a Snakefile that is outside of the stated working directory, we get a long, convoluted system path inside the .tar.gz and the test fails. If the snakefile is inside the working directory as it is expected, the test works as it should. This is the reason that I changed those lines. I'm not sure what else to do.

Reply:
I can change it back and the other test might pass again, but when you enable google life sciences testing it will fail. Let me know what you'd like to do...

Reply:
The test relies on the location being unchaged, indeed. What you describe for GLS is an important point to consider though. Is there a reason to not cleanup the path in your GLS code if the snakefile/workflow sources are outside of the workdir? I think that should be possible. Maybe just always move them into a place like `sources/`, while keeping the folder structure, but without the system specific common prefix.

Reply:
I could definitely do that - is `sources` a convention?

Reply:
@johanneskoester this is a deeper bug in snakemake - I can provide a path to a Snakefile somewhere else, and the correct working directory, but the workflow.included adds the wrong files:
```
self.workflow.get_sources()
Unable to retrieve additional files from git. This is not a git repository.
Out[1]: 
{'../../home/vanessa/Desktop/Code/snakemake/snakemake/tests/test_google_lifesciences/Snakefile',
 '../../home/vanessa/Desktop/Code/snakemake/snakemake/tests/test_google_lifesciences/env.yml'}
```
We can assume that the Snakefile could be an outside path from the working directory, but the rest of the included files should be derived from the working directory, no?

Reply:
This gets into very dangerous territory to start copying files around a user working directory - in fact we can't always assume that the user is okay with this.

Reply:
I don't know how to fix this because all the files in self.included are wrong.

Reply:
I think the best approach is the following:

 - put a warning to the user if a file it outside of the working directory and exit on fail. It's not up to snakemake to change the organization / move files on behalf of the user, and it's not unreasonable to require that the Snakefile is in the same workdir. This is akin to a Dockerfile, you can't build a container with any context below the Dockerfile.
 - the bug with the other job run needs to be fixed. It should not rely on the snakefile being in a specific (and arguably wrong) location.

Reply:
okay I've fixed the issue with the test! It was basically hard coding the original path for the file (the issue name) into the output folder, which doesn't make sense. It's running again now and I think we should be ok... sorry for the unneeded noise!